lib/chef/provider/google_key_pair.rb

```diff
@@ -41,20 +41,21 @@ class Chef::Provider::GoogleKeyPair < Chef::Provider::LWRPBase
   end
 
   action :destroy do
-    name_to_key = driver.project_client.get_ssh_mappings
+    metadata = driver.project_client.get
```

Comment:
Hmm. The point of this is to have one API call only to get all the metadata, modify it and then one call only to reupload it. I see your point and the previous version was a bit like what you described, but the previous version did one API call every time you changed something and I don't think that is desirable.